### DISCUSSION TITLE
Accessing form from external component outside context

### DISCUSSION THREAD

**Discussion by bombillazo** [react-hook-form/react-hook-form#8609] [Q&A] [answered]
Hello! First, im loving React Hook Forms!

However, I'm struggling with the following: how can I get form state/values outside of the component I am creating the form in? I thought about maybe using some ref to get the data but don't see any API for that with RHF, and I can't move the form outside to the parent without breaking my custom form structure.

**Comment by bluebill1049** [comment, marked as answer]
- Context usage will have to be used within the (React) context
- without Context usage will need to be within the props (pass down)
